src/io/g15.rs

```diff
@@ -0,0 +1,251 @@
+#![allow(non_upper_case_globals)]
+#![allow(unused_variables)]
+
+use regex::{Regex, Captures};
+use std::fmt;
+use std::fs;
+use std::num::ParseIntError;
+
+#[derive(Debug)]
+pub enum G15BadIndex {
+    IndexOutOfBounds, // Occurs when m_someArray[X] has some X greater than 33 (since these are static 34 element arrays)
+    BadInt,           // Occurs when an int fails to parse into the given unsigned range (u32 or u64)

```

Comment:
Since this is being used to propagate a `ParseIntError`, it should contain that error. e.g. `BadInt` -> `ParseError(ParseIntError)`. Then the From implementation for the error type would return `ParseError(value)` instead of just `BadInt`.

Reply:
Lets just momentarily forget the fact I forgor enums can represent concrete values

Reply:
Well tbf it is not as common in many other languages for enums to be able to hold data.

Reply:
I mean the two other languages I use (and write a lot in), Python and Java, both associate values with Enum constants. In fact, unless im mistaken or new Java versions allow this, Java doesn't even let you define naked enum elements

Reply:
Afaik enums in Java are just giving names to specific values of some type (usually just an int value), and can masquerade as classes, but the data any given enum can store must be the same type for all of the variants. Rust's enums are much more like Haskell's [Sum data types](https://www.schoolofhaskell.com/user/Gabriel439/sum-types).

With that you can have one variant of the enum which stores a `ParseIntError`, while the other variant contains no data, or you can add another vriant to store any type you want.